speechbrain/lobes/models/huggingface_wav2vec.py

```diff
@@ -39,162 +40,445 @@
-        self.output_norm = output_norm
         if self.freeze:
             logger.warning(
-                "speechbrain.lobes.models.huggingface_wav2vec - wav2vec 2.0 is frozen."
+                "speechbrain.lobes.models.HuggingFaceModel is frozen."
             )
             self.model.eval()
             for param in self.model.parameters():
                 param.requires_grad = False
         else:
+            self.model.gradient_checkpointing_disable()  # Required by DDP
             self.model.train()
-            if self.freeze_feature_extractor:
-                self.model.feature_extractor._freeze_parameters()
-        self.output_all_hiddens = output_all_hiddens
-
-    def _from_pretrained(self, source, config, model, save_path):
+            if freeze_nested_models_their_calls is not None:
+                if type(freeze_nested_models_their_calls) is list:
+                    freeze_nested_models_their_calls = [
+                        freeze_nested_models_their_calls
+                    ]
+                for nested_freeze_call in freeze_nested_models_their_calls:
+                    eval(f"self.model.{nested_freeze_call}()")
+
+    def _from_pretrained(
+        self,
+        source,
+        config,
+        model,
+        modify_state_dict_partial_fn,
+        save_path,
+        cache_dir,
+    ):
         """This function manages the source checking and loading of the params.
         # 1. Is the model from HF or a local path
         # 2. Is the model pretrained with HF or SpeechBrain
         # 3. Download (if appropriate) and load with respect to 1. and 2.
         """
-
-        is_sb, ckpt_file = self._check_model_source(source)
+        is_sb, ckpt_file = _check_model_source(source, save_path)
         if is_sb:
             config = config.from_pretrained(source, cache_dir=save_path)
             self.model = model(config)
             self.model.gradient_checkpointing_disable()  # Required by DDP
             # fetch the checkpoint file
             ckpt_full_path = fetch(
-                filename=ckpt_file, source=source, savedir=save_path
+                filename=ckpt_file,
+                source=source,
+                savedir=save_path,
+                cache_dir=cache_dir,
             )
             # We transfer the parameters from the checkpoint.
-            self._load_sb_pretrained_w2v2_parameters(ckpt_full_path)
+            self._load_sb_pretrained_parameters(
+                ckpt_full_path,
+                modify_state_dict_partial_fn=modify_state_dict_partial_fn,
+            )
         else:
             self.model = model.from_pretrained(source, cache_dir=save_path)
 
-    def _load_sb_pretrained_w2v2_parameters(self, path):
-        """Loads the parameter of a w2v2 model pretrained with SpeechBrain and the
-        HuggingFaceWav2Vec2Pretrain Object. It is necessary to perform a custom
+    def _load_sb_pretrained_parameters(
```

Comment:
same

Reply:
same - internal function of class - no args docs